File: backend/app/schemas/daily_task_completion.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import time


class DailyTaskCompletionBase(BaseModel):
    """日清表基础Schema"""
    
    # === 第一部分：关联任务信息 ===
    project_id: Optional[str] = Field(None, description="项目ID")
    project_name: Optional[str] = Field(None, description="项目名称")
    task_id: Optional[str] = Field(None, description="任务ID")
    task_name: Optional[str] = Field(None, description="任务名称")
    task_source: str = Field("custom", description="任务来源：my_tasks-我的任务, custom-自定义")
    
    # === 第二部分：工作内容 ===
    start_time: Optional[time] = Field(None, description="开始时间")
    end_time: Optional[time] = Field(None, description="结束时间")
    hours_spent: float = Field(0.0, description="实际耗时")
    planned_hours: float = Field(0.0, description="计划耗时")
    
    # 进度相关
    progress_status: str = Field("正常", description="进度状态：正常/延期/提前")
    progress_percentage: float = Field(0.0, description="完成百分比")
    delay_hours: float = Field(0.0, description="延期小时数")
    
    # 工作内容
    work_content: Optional[str] = Field(None, description="具体工作内容")
    key_work_tracking: Optional[str] = Field(None, description="重点工作跟踪")
    result: Optional[str] = Field(None, description="工作结果")
    measures: Optional[str] = Field(None, description="改进措施")
    
    # 评价相关
    self_evaluation: Optional[str] = Field(None, description="自评：A/B/C/D")
    supervisor_evaluation: Optional[str] = Field(None, description="主管评价：A/B/C/D")
    evaluation_comment: Optional[str] = Field(None, description="评价说明")
    
    # 明天计划
    tomorrow_plan: Optional[str] = Field(None, description="明天工作计划")
    
    # 状态
    status: str = Field("进行中", description="任务状态：进行中/已完成/暂停/取消")
    is_key_work: bool = Field(False, description="是否重点工作")


class DailyTaskCompletionCreate(DailyTaskCompletionBase):
    """创建日清表Schema"""
    report_id: int = Field(..., description="日报ID")


class DailyTaskCompletionUpdate(BaseModel):
    """更新日清表Schema"""
    
    # === 第一部分：关联任务信息 ===
    project_id: Optional[str] = Field(None, description="项目ID")
    project_name: Optional[str] = Field(None, description="项目名称")
    task_id: Optional[str] = Field(None, description="任务ID")
    task_name: Optional[str] = Field(None, description="任务名称")
    task_source: Optional[str] = Field(None, description="任务来源")
    
    # === 第二部分：工作内容 ===
    start_time: Optional[time] = Field(None, description="开始时间")
    end_time: Optional[time] = Field(None, description="结束时间")
    hours_spent: Optional[float] = Field(None, description="实际耗时")
    planned_hours: Optional[float] = Field(None, description="计划耗时")
    
    # 进度相关
    progress_status: Optional[str] = Field(None, description="进度状态")
    progress_percentage: Optional[float] = Field(None, description="完成百分比")
    delay_hours: Optional[float] = Field(None, description="延期小时数")
    
    # 工作内容
    work_content: Optional[str] = Field(None, description="具体工作内容")
    key_work_tracking: Optional[str] = Field(None, description="重点工作跟踪")
    result: Optional[str] = Field(None, description="工作结果")
    measures: Optional[str] = Field(None, description="改进措施")
    
    # 评价相关
    self_evaluation: Optional[str] = Field(None, description="自评")
    supervisor_evaluation: Optional[str] = Field(None, description="主管评价")
    evaluation_comment: Optional[str] = Field(None, description="评价说明")
    
    # 明天计划
    tomorrow_plan: Optional[str] = Field(None, description="明天工作计划")
    
    # 状态
    status: Optional[str] = Field(None, description="任务状态")
    is_key_work: Optional[bool] = Field(None, description="是否重点工作")


class DailyTaskCompletionResponse(BaseModel):
    """日清表响应Schema"""
    id: int
    report_id: int
    
    # === 第一部分：关联任务信息 ===
    project_id: Optional[str] = None
    project_name: Optional[str] = None
    task_id: Optional[str] = None
    task_name: Optional[str] = None
    task_source: str = "custom"
    
    # === 第二部分：工作内容 ===
    start_time: Optional[time] = None
    end_time: Optional[time] = None
    hours_spent: float = 0.0
    planned_hours: float = 0.0
    
    # 进度相关
    progress_status: str = "正常"
    progress_percentage: float = 0.0
    delay_hours: float = 0.0
    
    # 工作内容
    work_content: Optional[str] = None
    key_work_tracking: Optional[str] = None
    result: Optional[str] = None
    measures: Optional[str] = None
    
    # 评价相关
    self_evaluation: Optional[str] = None
    supervisor_evaluation: Optional[str] = None
    evaluation_comment: Optional[str] = None
    
    # 明天计划
    tomorrow_plan: Optional[str] = None
    
    # 状态
    status: str = "进行中"
    is_key_work: bool = False
    
    # 时间戳
    create_time: str
    update_time: str
    is_deleted: bool = False
    
    class Config:
        from_attributes = True


class DailyTaskCompletionListResponse(BaseModel):
    """日清表列表响应Schema"""
    items: List[DailyTaskCompletionResponse]
    total: int
    page: int
    size: int
    total_pages: int

File: frontend/src/api/dailyTaskCompletion.ts
import request from '@/api/axios'

// 日清表接口类型定义
export interface DailyTaskCompletion {
  id?: number
  report_id?: number
  project_id?: string
  project_name?: string
  task_id?: string
  task_name?: string
  task_source: string
  start_time?: string
  end_time?: string
  hours_spent: number
  planned_hours: number
  progress_status: string
  progress_percentage: number
  delay_hours: number
  work_content: string
  key_work_tracking?: string
  result?: string
  measures?: string
  self_evaluation?: string
  supervisor_evaluation?: string
  evaluation_comment?: string
  tomorrow_plan?: string
  status: string
  is_key_work: boolean
  create_time?: string
  update_time?: string
  is_deleted?: boolean
}

export interface DailyTaskCompletionCreate {
  report_id: number
  project_id?: string
  project_name?: string
  task_id?: string
  task_name?: string
  task_source?: string
  start_time?: string
  end_time?: string
  hours_spent?: number
  planned_hours?: number
  progress_status?: string
  progress_percentage?: number
  delay_hours?: number
  work_content: string
  key_work_tracking?: string
  result?: string
  measures?: string
  self_evaluation?: string
  supervisor_evaluation?: string
  evaluation_comment?: string
  tomorrow_plan?: string
  status?: string
  is_key_work?: boolean
}

export interface DailyTaskCompletionUpdate {
  id?: number
  report_id?: number
  project_id?: string
  project_name?: string
  task_id?: string
  task_name?: string
  task_source?: string
  start_time?: string
  end_time?: string
  hours_spent?: number
  planned_hours?: number
  progress_status?: string
  progress_percentage?: number
  delay_hours?: number
  work_content?: string
  key_work_tracking?: string
  result?: string
  measures?: string
  self_evaluation?: string
  supervisor_evaluation?: string
  evaluation_comment?: string
  tomorrow_plan?: string
  status?: string
  is_key_work?: boolean
}

export interface DailyTaskCompletionListResponse {
  items: DailyTaskCompletion[]
  total: number
  page: number
  size: number
  total_pages: number
}

export interface MyTaskItem {
  task_id: string
  task_name: string
  project_id: string
  project_name: string
}

export interface MyTaskResponse {
  items: MyTaskItem[]
  total: number
  message: string
}

export interface ReportSummary {
  total_hours: number
  completed_tasks: number
  ongoing_tasks: number
  key_works: number
  average_progress: number
}

export interface ReportSummaryResponse {
  report_id: number
  summary: ReportSummary
  message: string
}

// 创建日清表记录
export const createDailyTaskCompletion = (data: DailyTaskCompletionCreate): Promise<DailyTaskCompletion> => {
  return request.post<DailyTaskCompletion>('v1/daily-task-completion', data).then(response => {
    // 后端直接返回对象，不需要.data
    return response as any
  })
}

// 获取日清表记录列表
export const getDailyTaskCompletions = (params?: {
  report_date?: string
  start_date?: string
  end_date?: string
  status?: string
  page?: number
  size?: number
}): Promise<DailyTaskCompletionListResponse> => {
  return request.get<DailyTaskCompletionListResponse>('v1/daily-task-completion', { params }).then(response => {
    // 后端直接返回对象，不需要.data
    return response as any
  })
}

// 获取指定日报的日清表记录
export const getTaskCompletionsByReport = (reportId: number): Promise<DailyTaskCompletion[]> => {
  return request.get<DailyTaskCompletion[]>(`v1/daily-task-completion/report/${reportId}`).then(response => {
    // 后端直接返回数组，不需要.data
    return response as any
  })
}

// 获取日清表记录详情
export const getDailyTaskCompletion = (taskCompletionId: number): Promise<DailyTaskCompletion> => {
  return request.get<DailyTaskCompletion>(`v1/daily-task-completion/${taskCompletionId}`).then(response => {
    // 后端直接返回对象，不需要.data
    return response as any
  })
}

// 更新日清表记录
export const updateDailyTaskCompletion = (taskCompletionId: number, data: DailyTaskCompletionUpdate): Promise<DailyTaskCompletion> => {
  return request.put<DailyTaskCompletion>(`v1/daily-task-completion/${taskCompletionId}`, data).then(response => {
    // 后端直接返回对象，不需要.data
    return response as any
  })
}

// 删除日清表记录
export const deleteDailyTaskCompletion = (taskCompletionId: number): Promise<{ message: string }> => {
  return request.delete<{ message: string }>(`v1/daily-task-completion/${taskCompletionId}`).then(response => {
    // 后端直接返回对象，不需要.data
    return response as any
  })
}

// 批量创建日清表记录
export const bulkCreateTaskCompletions = (reportId: number, taskCompletions: DailyTaskCompletionCreate[]): Promise<DailyTaskCompletion[]> => {
  return request.post<DailyTaskCompletion[]>(`v1/daily-task-completion/report/${reportId}/bulk`, taskCompletions).then(response => {
    // 后端直接返回数组，不需要.data
    return response as any
  })
}

// 获取我的任务列表（用于任务选择器）
export const getMyTasksAvailable = (params?: {
  status?: string
  keyword?: string
  limit?: number
}): Promise<MyTaskResponse> => {
  return request.get<MyTaskResponse>('v1/daily-task-completion/my-tasks/available', { params }).then(response => {
    // 后端直接返回对象，不需要.data
    return response as any
  })
}

// 获取日报汇总信息
export const getReportSummary = (reportId: number): Promise<ReportSummaryResponse> => {
  return request.get<ReportSummaryResponse>(`v1/daily-task-completion/report/${reportId}/summary`).then(response => {
    // 后端直接返回对象，不需要.data
    return response as any
  })
}

// 快速添加日清表记录
export const quickAddTaskCompletion = (reportId: number, quickData: {
  work_content: string
  task_name?: string
  project_name?: string
  hours_spent?: number
  progress_percentage?: number
  status?: string
  is_key_work?: boolean
}): Promise<DailyTaskCompletion[]> => {
  return request.post<DailyTaskCompletion[]>(`v1/daily-task-completion/quick-add/${reportId}`, quickData).then(response => {
    // 后端直接返回数组，不需要.data
    return response as any
  })
}